components/ProductsTable/ProductsTable.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div class="overflow-x-auto">
+    <table class="table">
+      <!-- head -->
+      <thead>
+        <tr>
+          <th>Name</th>
+          <th>Stock</th>
+          <th>Price</th>
+          <th>Total Sales</th>
+        </tr>
+      </thead>
+      <tbody>
+        <!-- rows -->
+        <tr v-for="(user, index) in paginatedUsers" :key="index">
+          <td>
+            <div class="flex items-center gap-2">
+              <div class="image">
+                <img :src="user.image" :alt="`Image of ${user.name}`" />
+              </div>
+              <div>
+                <div class="font-bold">{{ user.name }}</div>
+                <div class="text-sm opacity-60">{{ user.description }}</div>
+              </div>
+            </div>
+          </td>
+          <td>{{ user.stock }}</td>
+          <td>{{ user.price }}</td>
+          <td>{{ user.totalSales }}</td>
+        </tr>
+      </tbody>
+      <!-- foot -->
+      <tfoot>
+        <!-- Pagination controls -->
+      </tfoot>
+    </table>
+  </div>
+</template>
+
+<script>
+import usersData from '~/products.json';
+
+export default {
+  data() {
+    return {
+      users: usersData,
+      currentPage: 1,
+      rowsPerPage: 4
+    };
+  },
+  computed: {
+    paginatedUsers() {
+      const startIndex = (this.currentPage - 1) * this.rowsPerPage;
+      const endIndex = startIndex + this.rowsPerPage;
+      return this.users.slice(startIndex, endIndex);
+    }
+  }
+};
+</script>
```

Comment:
instead of 
```
<script>
...
</script>
```
You should use:

```
<script setup>
defineProps({
  currentPage: {
    type: Number,
    default: () => 1,
  },
  rowsPerPage: {
    type: Number,
    default: () => 4,
  },
});
const data = ref([]);

const fetchData = async () => {
  //timespan to simulate waiting for api response
  await new Promise((resolve) => setTimeout(resolve, 1000));

  const response = await $fetch("/api/products", {
    method: "GET",
  });
  return response;
};

data.value = await fetchData();
</script>
```

With params You can reuse component on other pages